packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +70,261 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-        if fns.len() != 3 {
+        // Double braces like `{{` are Rust's escape string for a single literal `{`.
+        let wrapped_code = new_js_string(
+            scope,
+            &format!(
+                "((hash_util, hash_stdlib) => {{{code}
+                    return [
+                        typeof start_experiment === 'undefined' ? undefined : start_experiment,
+                        typeof start_sim === 'undefined' ? undefined : start_sim,
+                        typeof run_task === 'undefined' ? undefined : run_task
+                    ]
+                }})",
+            ),
+        )?;
+
+        let pkg: v8::Local<'_, v8::Function> = {
+            let pkg = v8::Script::compile(scope, wrapped_code, None)
+                .ok_or_else(|| Error::V8("Could not create wrapped code scipt".to_string()))?;
+            let pkg = pkg
+                .run(scope)
+                .ok_or_else(|| Error::V8("Could not execute wrapped code script".to_string()))?;
+
+            pkg.try_into().map_err(|err| {
+                Error::V8(format!("Could not convert wrapped code to Function: {err}"))
+            })?
+        };
+
+        let args = &[embedded.hash_util, embedded.hash_stdlib];
+        let fns: v8::Local<'_, v8::Array> = {
+            let global_context = context.global(scope);
+            let fns = pkg
+                .call(scope, global_context.into(), args)
+                .ok_or_else(|| {
+                    Error::PackageImport(
+                        path.clone(),
+                        "Could not run wrapped code function".to_string(),
+                    )
+                })?;
+
+            fns.try_into().map_err(|err| {
+                Error::V8(format!(
+                    "Could not convert wrapped code return value to Array: {err}"
+                ))
+            })?
+        };
+
+        if fns.length() != 3 {
             return Err(Error::PackageImport(path.clone(), "Stray return".into()));
         }
 
         // Validate returned array.
         let fn_names = ["start_experiment", "start_sim", "run_task"];
-        for (elem, fn_name) in fns.clone().elements().zip(fn_names) {
-            let elem: mv8::Value<'_> = elem.map_err(|e| {
-                Error::PackageImport(path.clone(), format!("Couldn't index array: {:?}", e))
-            })?;
+        for (idx_fn, fn_name) in fn_names.iter().enumerate() {
+            let elem: v8::Local<'_, v8::Value> =
+                fns.get_index(scope, idx_fn as u32).ok_or_else(|| {
+                    Error::PackageImport(path.clone(), format!("Couldn't index array: {idx_fn:?}"))
+                })?;
+
             if !(elem.is_function() || elem.is_undefined()) {
                 return Err(Error::PackageImport(
                     path.clone(),
-                    format!("{} should be a function, not {:?}", fn_name, elem),
+                    format!("{fn_name} should be a function, not {elem:?}"),
```

Comment:
```suggestion
                    format!("{fn_name} returned from package {name} should be a function, not {elem:?}"),
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/2a832f8dc26357b06320aed8e66cb59822d849a2